added support for rdf site summary (rss 1.0)

hamtie opened the pull request:
e.g. http://sfbay.craigslist.org/ccc/index.rss can now be parsed with the rsshandler

fb55 committed:
Restructured code, improved readability, deleted junk

fb55 committed:
didn't export constructors

fb55 committed:
Now all tests pass

fb55 committed:
moved runtests.js to tests-directory

fb55 committed:
fixed tests

fb55 committed:
moved "last()"-method from _callStack to DefaultHandlers prototype

fb55 committed:
removed repeating code in RssHandler

fb55 committed:
again some cleanup

fb55 committed:
Added EventedHandler, using an interface like sax.js

fb55 committed:
Some improvements inside the parser. Still very ugly.

fb55 committed:
Export the evented handler

fb55 committed:
Fixed a bug & restructured some code

fb55 committed:
Wrong number of arguments was passed in EventedHandler

fb55 committed:
Removed repeating code

fb55 committed:
renamed module, added it to npm (as "htmlparser2")

fb55 committed:
Replaced indexOf(a)!==0 with substring(a.length)!==a, charAt(length-1) with substr(-1) + many other improvements

fb55 committed:
1.1.0

fb55 committed:
Restructured DomUtils

fb55 committed:
Some small improvements

fb55 committed:
npm bugfix

fb55 committed:
use util.inherits (function was taken from node anyway)

fb55 committed:
Code cleanup, preparation for big changes

fb55 committed:
Create element object later

fb55 committed:
Prepared big changes

The element stack will be removed, a new event structure will be
introduced

fb55 committed:
Some steps preparing for the upcoming changes

fb55 committed:
Removed regexp for comment trimming

use String#slice instead

fb55 committed:
Deleted browser related content in tests

fb55 committed:
Added a test

https://github.com/tautologistics/node-htmlparser/issues/29

fb55 committed:
Fixed test 23

fb55 committed:
Removed trash in runtests.js

fb55 committed:
Replaced _reTags with indexOf

fb55 committed:
Added doctype to first test

fb55 committed:
Removed char loop from Parser#getLocation

Use String#split & String#replace instead

fb55 committed:
Renamed htmlparser.js to index.js

The old name was misleading.

fb55 committed:
Don't allow <tagname <tagname

fb55 committed:
corrected reference in runtests

fb55 committed:
Small changes to Parser#getLocation

fb55 committed:
Updated readme

fb55 committed:
readme, again

fb55 committed:
Added lines to readme

fb55 committed:
removed backticks from readme

fb55 committed:
Removed last try-catch-blocks from RssHandler

fb55 committed:
parseAttributes now returns an object

fb55 committed:
parseAttributes now accepts an elements data

fb55 committed:
minor improvements

fb55 committed:
Restructured some code

fb55 committed:
bugfix

fb55 committed:
Small adjustments

hamtie committed:
added support for rss feeds in rdf format

fb55 committed:
1.2.0

fb55 committed:
Added RDF feed support to RssHandler

Idea & part of code from
https://github.com/tautologistics/node-htmlparser/pull/35

fb55 commented:
Great idea. I merged it into my own [fork](https://github.com/FB55/node-htmlparser). (Besides, I changed the code to have a better readable coding style.)

fb55 committed:
restructured code

fb55 committed:
Removed duplicated code

fb55 committed:
Splited writeTag in both handlers to smaller functions + added a test for events

fb55 committed:
added a limit to the elements fetched inside RssHandler

fb55 committed:
improved DomUtils

what I really wanted to do is now in comments (it failed multiple times)

fb55 committed:
some small adjustments

fb55 committed:
Replaced _tagStack with _contentFlags, tweaked DefaultHandler

That fixed https://github.com/tautologistics/node-htmlparser/issues/29.

fb55 committed:
Made openTag & closeTag public methods

Just deleted the "_" in front of them

fb55 committed:
Improved _addDomElement

fb55 committed:
Set feeds to xmlMode

because they are xml, so <script> & <style> tags shouldn't get a
special treatment

fb55 committed:
minor changes

fb55 committed:
Removed _elements, removed support for writeTag inside handlers

Use openTag and closeTag instead

fb55 committed:
RssHandler now calls a given callback

fb55 committed:
updated readme

fb55 committed:
fixed links in readme

fb55 committed:
Updated readme

fb55 committed:
Minor changes

fb55 committed:
Added options to EventedHandler

for now, it's only enforceEmptyTags

fb55 committed:
fixed a potential bug

fb55 committed:
adjustments

hamtie committed:
Merge branch 'master' of git://github.com/FB55/node-htmlparser

hamtie committed:
rss parsing can grab all tags now. todo: write a normalization
function for disparing names in different rss types.

hamtie commented:
cool. I just merged your branch into mine.  My last commit parses the feeds a little more dynamically.  This undoes some of the normalization of the names, e.g. "lastBuildDate" is no longer returned as "updated" but I'm doing normalization after parsing it anyway.  I might add a convenience function that automatically normalizes some of the crucial names later.

fb55 commented:
I like the idea to not check every single item multiple times, but normalisation is absolutely required. My idea is to rewrite the entire thing to use my EventedHandler, so that it's not necessary to build a DOM or anything.

Besides: I already opened a pull request for my own changes, so if you want to change my code, it would be easier if you fork my fork and send me a pull requests. I don't think @tautologistics is still actively developing this project, so any pull requests here will probably be forgotten.
